Create release notes for release 2.1 (#839)
kyma-project/website: content/blog-posts/2022-03-24-release-notes-2.1/index.md

- [Service Mesh](#service-mesh) - FIPS distroless Istio images provided by Solo, Istio refactored and upgraded to 1.12.3

## General

With release 2.1, we now officially support and test against Kubernetes 1.21.

## CLI

### The `deploy` command: try the new `dry-run` option in an alpha version

With this release, a new `dry-run` option was added to the `deploy` command. It lets you trial-run the YAMLs for your Kubernetes resources and see what the output would be if they were deployed. Mind that for now, the option is in the alpha version. See the [Kyma CLI documentation](https://github.com/kyma-project/cli/blob/main/docs/gen-docs/kyma_deploy.md) for more details.

### The `deploy` command: specify components with **downloadURL** and **version**

Another addition to the `deploy` command is an extension to the `--component` flag. Previously you could specify the **name** and **namespace** for the component. Now you can also specify the two additional parameters: **downloadURL** and **version**. Again, refer to the [CLI documentation](https://github.com/kyma-project/cli/blob/main/docs/gen-docs/kyma_deploy.md) for more details. 

### Generate schema for Function manifests

> We improved development experience for Functions developers. We added schema generation for the `config.yaml` manifests. This, when combined with dedicated plugins in your IDE, helps you avoid mistakes via validation and auto-completion.

Reviewer: ```suggestion
We improved the development experience for Functions developers. We added schema generation for the `config.yaml` manifests. This, when combined with dedicated plugins in your IDE, helps you avoid mistakes via validation and auto-completion.
```

> - If you use the VSCode IDE to develop Function code, just  use the `--vscode` option when generating the Function project:

Reviewer: ```suggestion
- If you use the VSCode IDE to develop Function code, just use the `--vscode` option when generating the Function project:
```

  ```bash
  kyma init function --vscode
  ```

>   This command  creates the `.vscode` folder, which contains the schema for `config.yaml`. Install a [YAML plugin](https://github.com/redhat-developer/vscode-yaml) in your IDE and the VSCode will automatically pick it up for validation and autocompletion.

Reviewer: ```suggestion
  This command  creates the `.vscode` folder, which contains the schema for `config.yaml`. Install a [YAML plugin](https://github.com/redhat-developer/vscode-yaml) in your IDE and the VSCode IDE will automatically pick it up for validation and autocompletion.
```

Author: I wouldn't repeat it here, I think it's redundant. Everyone knows what VSCode is, plus line 46 says specifically that we're talking about the IDE. Is it okay with you?

Reviewer: Fully agree, but then we don't need the article imho 
```suggestion
  This command  creates the `.vscode` folder, which contains the schema for `config.yaml`. Install a [YAML plugin](https://github.com/redhat-developer/vscode-yaml) in your IDE and VSCode will automatically pick it up for validation and autocompletion.
```

- If you just want to generate the schema, use the following command:

  ```bash
  kyma get schema serverless
  ```

  This will print out the JSON schema so that you can import it in your preferred IDE.

## Eventing

### Support for non-alphanumeric characters in event types

The naming in event types comes with certain restrictions.

> Previously, we only supported ASCII alphanumeric characters from the range `[a-zA-Z0-9]+`. If your name contained an offensive character, the event sending was failed.

Reviewer: I know what you mean, but my first association was swear words. How about:

```suggestion
Previously, we only supported ASCII alphanumeric characters from the range `[a-zA-Z0-9]+`. If your name contained an unsupported character, the event sending was failed.
```

> Starting from this release, we no longer fail sending such events. Instead, unsupported characters are automatically dropped from events sent to Kyma and also from event types specified in the Subscription CRD.

Reviewer: If you want to avoid repetition, we could say "invalid" here. But personally, I think repetition would be fine.

```suggestion
Starting from this release, we no longer fail sending such events. Instead, invalid characters are automatically dropped from events sent to Kyma and also from event types specified in the Subscription CRD.
```

Read about the [event names](https://kyma-project.io/docs/kyma/2.1/05-technical-reference/evnt-01-event-names) and [Subscription CRD](https://kyma-project.io/docs/kyma/2.1/05-technical-reference/00-custom-resources/evnt-01-subscription) in more detail.

## Observability

### Kiali and Jaeger upgraded

With Kyma 2.1, the Kiali and Tracing components got updated to the following recent versions:
- Kiali 1.44
- Jaeger 1.30.0

### Monitoring upgraded

Speaking of upgrades, the different components of the monitoring area also got one. While doing that, we felt inspired and updated the used Helm chart, too, and adapted it to the upstream charts. The components were upgraded to the following versions:

- Node-exporter 1.3.1
- Kube-state-metrics 2.3.0
- Prometheus 2.32.1
- Prometheus-operator 0.53.1
- Alertmanager 0.23.0
- Pushgateway 1.4.2
- Grafana 7.5.15
- Oauth2-proxy 7.2.1

### Metrics restructured

> As the kube-state-metrics component got a major upgrade to version 2.x, some metrics got restructured. 

Reviewer: ```suggestion
As the kube-state-metrics component got a major upgrade to version 2.x, some metrics have been restructured. 
```

Author: I'd stick to `got`, as it's part of the same change, so I'd apply the same tense. Or I can change it to `were` if you insist on replacing `got` with something. WDYT?

Reviewer: I'd opt for "were" then :) 
I think the [get passive](https://www.grammaring.com/the-passive-with-get) is too informal for release notes.

```suggestion
As the kube-state-metrics component got a major upgrade to version 2.x, some metrics were restructured. 
```

Author: I doesn't strike me as such, as in the OS Kyma we try to keep the style not-so-formal, but of course there's no harm in changing it to "were". :) 
But just to clarify, the "informal" use only applies to the first example (first 2 sentences) from the link you provided, the remaining ones are actually not informal use.
<img width="806" alt="image" src="https://user-images.githubusercontent.com/53601578/159881615-f8b5c13a-027d-4ed7-8d86-60fc8f153c1f.png">

 "Get" expresses, that there was a change, which is exactly what the release notes are about :)

The following table shows how the affected labels were restructured:

| Previously  | Now |
|-------------|-----|
| `kube_pod_container_resource_requests_cpu_cores`, <br /> `kube_pod_container_resource_limits_cpu_cores`, <br /> `kube_pod_container_resource_requests_memory_bytes`, <br /> `kube_pod_container_resource_limits_memory_byte`            | `kube_pod_container_resource_requests`, <br /> `kube_pod_container_resource_limits`   |
| `kube_node_status_capacity_pods`, <br /> `kube_node_status_capacity_cpu_cores`, <br /> `kube_node_status_capacity_memory_bytes`, <br /> `kube_node_status_allocatable_pods`, <br /> `kube_node_status_allocatable_cpu_cores`, <br /> `kube_node_status_allocatable_memory_bytes`,            | `kube_node_status_capacity`, <br /> `kube_node_status_allocatable`   |
| `kube_daemonset_updated_number_scheduled`            | `kube_daemonset_status_updated_number_scheduled`    |
| `kube_hpa_*`            | `kube_horizontalpodautoscaler_*`    |

> Furthermore, the `envoy_` metrics exposed by the istio-sidecar running with every workload are not getting collected by default anymore. 

Reviewer: ```suggestion
Furthermore, the `envoy_` metrics exposed by the istio-sidecar running with every workload are not collected by default anymore. 
```

> However, we continue to collect the more relevant `istio_` metrics which are used in the Istio-specific dashboards. 

Reviewer: ```suggestion
However, we continue to collect the more relevant `istio_` metrics, which are used in the Istio-specific dashboards. 
```

Author: I wasn't sure here if it was talking about a subset of Istio metrics (no comma) or all of the Istio metrics (comma). Need to check with @strekm .

Author: Adding the comma.

It turned out that the `envoy_` metrics were helpful only in advanced troubleshooting scenarios and introduced too big a cardinality, resulting in big resource consumption. Still, you can enable them as optional. 
See [#issue 13659](https://github.com/kyma-project/kyma/issues/13659) for more details.

### Grafana dashboards improved

> Several improvements were applied to the dashboards coming with the Grafana in Kyma.

Reviewer: ```suggestion
Several improvements were applied to the dashboards that come with the Grafana in Kyma.
```

> - The Loki dashboard got improved by adding a new top section with the most relevant metrics displayed.

Reviewer: ```suggestion
- The Loki dashboard has been improved by adding a new top section with the most relevant metrics displayed.
```

Reviewer: then I'd prefer "was" :)

Author: Already changed that yesterday :)

- All dashboards were adjusted to the metric changes coming with the major upgrade of kube-state-metrics.
- The dashboards tagged as `kubernetes-mixins` were updated to the latest available version from the community.
- `node-exporter` and Prometheus dashboards were replaced by the version provided by the community.
- The Kyma `frontends` and `backends` dashboards were removed, as they had no additional value to existing dashboards.
- The Kyma `Controllers` dashboards were revamped to have relevant reconciliation metrics included, and they were rolled out to all Kyma controllers.

### The `alertmanager` chart revamped

> The Kyma `alertmanager` chart, which is a sub-chart of the monitoring component, got refactored more heavily. Mainly, we tried to remove customizations that were in contrast to the available community chart, to simplify future updates.

Reviewer: ```suggestion
The Kyma `alertmanager` chart, which is a sub-chart of the monitoring component, was refactored more heavily. Mainly, we tried to remove customizations that were in contrast to the available community chart, to simplify future updates.
```

With that, the following changes were applied:

- The dedicated configuration options for VictorOps were removed. Use the typical options for configuration of notification channels instead.
- New alert rules were added.
  - `AlertmanagerFailedToSendAlerts`
  - `AlertmanagerClusterFailedToSendAlerts`
  - `AlertmanagerClusterFailedToSendAlerts Warning`
  - `AlertmanagerClusterDown`
  - `AlertmanagerClusterCrashlooping`
  - `ConfigReloaderSidecarErrors`
  - `KubeStateMetricsShardingMismatch`
  - `KubeStateMetricsShardsMissing`
  - `KubeAPITerminatedRequests`
  - `NodeFileDescriptorLimit`
  - `PrometheusLabelLimitHit`
  - `PrometheusTargetSyncFailure`
  - `PrometheusOperatorSyncFailed`
- The following alert rules were renamed:
  - `AggregatedAPIDown` → `KubeAggregatedAPIDown`
  - `AggregatedAPIErrors` → `KubeAggregatedAPIErrors`

### Logging chart updated and Loki improved

We updated the logging component to the latest available FluentBit version 1.8.13.

> Furthermore, the label map for the Loki integration got improved:

Reviewer: ```suggestion
Furthermore, the label map for the Loki integration has been improved:
```

Author: See above.

Reviewer: "was" ? :)

Author: Already change this one yesterday as well :)

- Previously, we had the Kubernetes `app` label mapped to the Loki `app` label. Now, on top of that, we also mapped the label `app.kubernetes.io/name` to the Loki `app` label.
- We mapped the label `app.kubernetes.io/component` to the Loki `component` label.
- We removed the `release` and `serverless.kyma-project.io/uuid` labels to improve the chunk size handling in Loki.

> Moreover, besides having data retention **by time**, we introduced a new sidecar taking care of data retention **by size**. With that, Loki will not crash anymore when reaching the volume size limit. Oldest data gets deleted first.

Reviewer: ```suggestion
Moreover, besides having data retention **by time**, we introduced a new sidecar taking care of data retention **by size**. With that, Loki will not crash anymore when reaching the volume size limit. Oldest data is deleted first.
```

For now, we recommend that you re-configure all your Node.js 12 Functions to run on the Node.js 14 runtime.

## Service Mesh

### FIPS distroless Istio images provided by Solo

With the Kyma 2.1, both [Istio control plane and data plane](https://istio.io/latest/docs/ops/deployment/architecture/) now use distroless images compliant with Federal Information Processing Standards (FIPS). [Solo](https://www.solo.io/) provides the FIPS-certified images. To learn more, read about [Distroless FIPS-compliant Istio](https://www.solo.io/blog/distroless-fips-compliant-istio/).

### Istio upgraded to 1.12.3

With this release, we upgraded Istio from 1.11.4 to 1.12.3. For more details on the introduced changes, read the official [Istio 1.12.3 release notes](https://istio.io/latest/news/releases/1.12.x/announcing-1.12.2/).

### Istio refactored

> With Kyma 2.1, we improved the Istio component, splitting it into Istio and Istio Resources charts. The Istio component is now solely responsible for pure Istio installatoion. The Istio Resources chart includes Kyma-specific configuration.

Reviewer: ```suggestion
With Kyma 2.1, we improved the Istio component, splitting it into Istio and Istio Resources charts. The Istio component is now solely responsible for pure Istio installation. The Istio Resources chart includes Kyma-specific configuration.
```

---
title: "Kyma 2.1"
author:
  name: "Maja Kurcius, Technical Writer @Kyma"
tags:
  - release-notes 
type: release
releaseTag: "2.1.0"
redirectFrom:
  - "/blog/release-notes-21"
---

> Ahoy, mateys! We're calling at the port again to load up with a fresh portion of new features, improvements, and upgrades, but also to unload stuff that we no longer need on this journey and that's been slowing us down. Support for Kubernetes 1.21, a new alpha option for the `deploy` command, support for non-alphanumeric characters in event types, restructure of metrics, and Istio refactorization are just some of those things that we happily bring aboard. Read on to find out more about this Kyma 2.1 ship!

Author: Actually, I wondered about it too, so I researched that and came across [this](https://wikidiff.com/refactorization/refactoring). Alas, now that I read it again, I realized I misunderstood it. 
I took 
> (...) refactorization is a split into constituent parts after a previous combination.

as meaning "after previously refactoring the code" 🤦‍♀️ 
Thanks for bringing that up. Changing it right away.

<!-- overview -->

> > **CAUTION:** Before upgrading to Kyma 2.1, read the [Migration Guide](https://kyma-project.io/docs/kyma/2.1/migration-guide-2.0-2.1).

Author: Good catch! [Done](https://github.com/kyma-project/kyma/pull/13715).

See the overview of all changes in this release:

- [General](#general) - Support for Kubernetes 1.21.
- [CLI](#cli) - Dry-run option for the `deploy` command in alpha, specifying components with **downloadURL** and **version**, generating schema for Function manifests, support for Gardener cluster hibernation, provisioning commands for GKE, AKS, and AWS removed
- [Eventing](#eventing) - Support for non-alphanumeric characters in event types
- [Observability](#observability) - Kiali and Jaeger upgraded, monitoring upgraded, metrics restructured, Grafana dashboards improved, the `alertmanager` chart revamped, logging chart updated and Loki improved
- [Serverless](#serverless) - Node.js 12 deprecated
- [Service Mesh](#service-mesh) - FIPS distroless Istio images provided by Solo, Istio refactored and upgraded to 1.12.3

## General

With release 2.1, we now officially support and test against Kubernetes 1.21.

## CLI

### The `deploy` command: try the new `dry-run` option in an alpha version

> With this release, a new `dry-run` option was added to the `deploy` command. It lets you trial-run the YAMLs for your Kubernetes resources and see what the output would be if they were deployed. Mind that for now, the option is in the alpha version. See the [Kyma CLI documentation](https://github.com/kyma-project/cli/blob/main/docs/gen-docs/kyma_deploy.md) for more details.

Author: The YAMLs for your Kubernetes resources in Kyma. I'd leave `the` here. Also, I prefer to leave `and` here. Leaving as is.

### The `deploy` command: specify components with **downloadURL** and **version**

Another addition to the `deploy` command is an extension to the `--component` flag. Previously you could specify the **name** and **namespace** for the component. Now you can also specify the two additional parameters: **downloadURL** and **version**. Again, refer to the [CLI documentation](https://github.com/kyma-project/cli/blob/main/docs/gen-docs/kyma_deploy.md) for more details. 

### Generate schema for Function manifests

We improved the development experience for Functions developers. We added schema generation for the `config.yaml` manifests. This, when combined with dedicated plugins in your IDE, helps you avoid mistakes via validation and auto-completion.

- If you use the VSCode IDE to develop Function code, just use the `--vscode` option when generating the Function project:
  ```bash
  kyma init function --vscode
  ```

>   This command  creates the `.vscode` folder, which contains the schema for `config.yaml`. Install a [YAML plugin](https://github.com/redhat-developer/vscode-yaml) in your IDE and VSCode will automatically pick it up for validation and autocompletion.

Author: Oh, you're totally right! Done.

- If you just want to generate the schema, use the following command:

  ```bash
  kyma get schema serverless
  ```

  This will print out the JSON schema so that you can import it in your preferred IDE.

### Support for Gardener cluster hibernation added

Starting with Kyma 2.1, it is now possible to hibernate a Kyma cluster which was created via Gardener. To do so, specify a hibernation schedule by providing [cron expressions](https://en.wikipedia.org/wiki/Cron) for the start and end of the hibernation, and the timezone on which to base the schedule:

```bash
kyma provision gardener gcp -n my-cluster -p my-project -c /path/to/credentials.yaml -s gcp-secret --hibernation-start="00 18 * * 1,2,3,4,5" hibernation-end="00 09 * * 1,2,3,4,5" hibernation-location="Europe/Berlin" 
``` 

> By default, all Gardener clusters have hibernation scheduled for weekdays at 6pm (18:00).

Author: The timezone is listed in the command (`hibernation-location`). The additional explanation is so that it's easier to decipher the cron expression in `--hibernation-start` and `--hibernation-end` without navigating to the link. 
Do you think we still need to add the timezone to the explanation?

> By default, all Gardener clusters have hibernation scheduled for weekdays at 6pm (18:00).

Author: Also, we should not write CET, as this changes throughout the year to CEST. But I'll include the timezone :)

### Provisioning commands for GKE, AKS, and AWS removed

> With this release, we removed the support for provisiong Kyma clusters on the GKE, AWS, and AKS hyperscalers. Provisioning of a cluster via Gardener or k3d is still possible.

Author: Need to ask @tobiscr as this was added to the input yesterday, but it's a good idea.

> With this release, we removed the support for provisiong Kyma clusters on the GKE, AWS, and AKS hyperscalers. Provisioning of a cluster via Gardener or k3d is still possible.

Reviewer 2: Hi @klaudiagrz,
I can answer the CLI related release notes. The main reason for this is dropping terraform so we are no longer blocked from updating dependencies on the CLI which was causing lots of issues moving forward (e.g. not being able to use latest kubernetes APIs).

From a customer perspective we could say that kyma does not officially support provisioning on 3rd party cloud providers anymore. You can still use the bring your own cluster model to install kyma on those providers, but they are not part of the supported environments (which are gardener and k3d).

> With this release, we removed the support for provisiong Kyma clusters on the GKE, AWS, and AKS hyperscalers. Provisioning of a cluster via Gardener or k3d is still possible.

Author: Thank you for the explanation!

> With this release, we removed the support for provisiong Kyma clusters on the GKE, AWS, and AKS hyperscalers. Provisioning of a cluster via Gardener or k3d is still possible.

Reviewer 3: Right, let's sell it in the way how @clebs proposed it.

We shouldn't disclosure that the continuous maintenance for lib-updates and especially the massive security impacts we had were no longer acceptable.

## Eventing

### Support for non-alphanumeric characters in event types

The naming in event types comes with certain restrictions.
Previously, we only supported ASCII alphanumeric characters from the range `[a-zA-Z0-9]+`. If your name contained an unsupported character, the event sending was failed.

> Starting from this release, we no longer fail sending such events. Instead, the unsupported characters are automatically dropped from events sent to Kyma and also from event types specified in the Subscription CRD.

Reviewer 4: What happens when the unsupported characters are automatically dropped but another event already has such a name? For example, we have two events: `event` and `event%`. So after the automatic dropping of the unsupported characters, both events would have to be named as `event`, which leads to a conflict. What then? 🤔

Author: I don't think there would be a conflict, as these are not resources, but events that are logged, so my understanding is it will just be logged/discovered as `event`, both of them, so yeah, it might lead to confusion, but not conflicts. I'll double check with @k15r though.

Reviewer 5: That problem is known and also mentioned in the kyma documentation:

> This could lead to a naming collision. For example, both system-prod and systemprod become systemprod. While this won't result in an error, it can cause Kyma to not work as expected. (https://kyma-project.io/docs/kyma/main/05-technical-reference/evnt-01-event-names#event-name-cleanup)

Reviewer 5: on our side there is no way to avoid these collisions 😢

Reviewer 4: ^ Maybe we can also mention that explicitly in the release notes? 🤔

Author: We already link to the documentation where it's mentioned, I think it's enough.

Read about the [event names](https://kyma-project.io/docs/kyma/2.1/05-technical-reference/evnt-01-event-names) and [Subscription CRD](https://kyma-project.io/docs/kyma/2.1/05-technical-reference/00-custom-resources/evnt-01-subscription) in more detail.

## Observability

### Kiali and Jaeger upgraded

> With Kyma 2.1, the Kiali and Tracing components were updated to the following recent versions:

Reviewer 4: ```suggestion
With Kyma 2.1, the Kiali and Jaeger components were updated to the following recent versions:
```

Author: I also wondered about it, but did not question it. I guess I was just under pressure to be finished by today. 🤷‍♀️ 
I'll double check if this was intentional and if not, I'll change it. Thanks!

- Kiali 1.44
- Jaeger 1.30.0

### Monitoring upgraded

Speaking of upgrades, the different components of the monitoring area also got one. While doing that, we felt inspired and updated the used Helm chart, too, and adapted it to the upstream charts. The components were upgraded to the following versions:

- Node-exporter 1.3.1
- Kube-state-metrics 2.3.0
- Prometheus 2.32.1

> - Prometheus-operator 0.53.1

Reviewer 4: ```suggestion
- Prometheus Operator 0.53.1
```

- Alertmanager 0.23.0
- Pushgateway 1.4.2
- Grafana 7.5.15

> - Oauth2-proxy 7.2.1

Reviewer 4: ```suggestion
- Oauth2 Proxy 7.2.1
```

Reviewer 4: That's how they are called in the official docs.

Author: I think it's written like this to refer to the name of the component. (If so, they should be in code font, but aaah, it's a hassle to distinguish these sometimes 🙈) I'll get back to this.

Author: Done.

### Metrics restructured

As the kube-state-metrics component got a major upgrade to version 2.x, some metrics were restructured. 

The following table shows how the labels were affected:

| Previously  | Now |
|-------------|-----|
| `kube_pod_container_resource_requests_cpu_cores`, <br /> `kube_pod_container_resource_limits_cpu_cores`, <br /> `kube_pod_container_resource_requests_memory_bytes`, <br /> `kube_pod_container_resource_limits_memory_byte`            | `kube_pod_container_resource_requests`, <br /> `kube_pod_container_resource_limits`   |
| `kube_node_status_capacity_pods`, <br /> `kube_node_status_capacity_cpu_cores`, <br /> `kube_node_status_capacity_memory_bytes`, <br /> `kube_node_status_allocatable_pods`, <br /> `kube_node_status_allocatable_cpu_cores`, <br /> `kube_node_status_allocatable_memory_bytes`,            | `kube_node_status_capacity`, <br /> `kube_node_status_allocatable`   |
| `kube_daemonset_updated_number_scheduled`            | `kube_daemonset_status_updated_number_scheduled`    |
| `kube_hpa_*`            | `kube_horizontalpodautoscaler_*`    |

> Furthermore, the `envoy_` metrics exposed by the istio-sidecar running with every workload are not collected by default anymore. 

Reviewer 4: ```suggestion
Furthermore, the `envoy_` metrics exposed by the Istio sidecar running with every workload are not collected by default anymore. 
```

However, we continue to collect the more relevant `istio_` metrics, which are used in the Istio-specific dashboards. 
It turned out that the `envoy_` metrics were helpful only in advanced troubleshooting scenarios and introduced too big a cardinality, resulting in big resource consumption. Still, you can enable them optionally. 
See [#issue 13659](https://github.com/kyma-project/kyma/issues/13659) for more details.

### Grafana dashboards improved

> Several improvements were applied to the dashboards that come with Grafana in Kyma.

Reviewer 4: ```suggestion
Several improvements were applied to the dashboards that come with Grafana in Kyma:
```

- The Loki dashboard was improved by adding a new top section with the most relevant metrics displayed.
- All dashboards were adjusted to the metric changes coming with the major upgrade of kube-state-metrics.
- The dashboards tagged as `kubernetes-mixins` were updated to the latest available version from the community.
- `node-exporter` and Prometheus dashboards were replaced by the version provided by the community.
- The Kyma `frontends` and `backends` dashboards were removed, as they had no additional value to existing dashboards.
- The Kyma `Controllers` dashboards were revamped to have relevant reconciliation metrics included, and they were rolled out to all Kyma controllers.

### The `alertmanager` chart revamped

The Kyma `alertmanager` chart, which is a sub-chart of the monitoring component, was refactored more heavily. Mainly, we tried to remove customizations that were in contrast to the available community chart, to simplify future updates.

With that, the following changes were applied:

- The dedicated configuration options for VictorOps were removed. Use the typical options for configuration of notification channels instead.

> - New alert rules were added.

Reviewer 4: ```suggestion
- New alert rules were added:
```

Author: Good catch, thanks!

  - `AlertmanagerFailedToSendAlerts`
  - `AlertmanagerClusterFailedToSendAlerts`
  - `AlertmanagerClusterFailedToSendAlerts Warning`
  - `AlertmanagerClusterDown`
  - `AlertmanagerClusterCrashlooping`
  - `ConfigReloaderSidecarErrors`
  - `KubeStateMetricsShardingMismatch`
  - `KubeStateMetricsShardsMissing`
  - `KubeAPITerminatedRequests`
  - `NodeFileDescriptorLimit`
  - `PrometheusLabelLimitHit`
  - `PrometheusTargetSyncFailure`
  - `PrometheusOperatorSyncFailed`
- The following alert rules were renamed:
  - `AggregatedAPIDown` → `KubeAggregatedAPIDown`
  - `AggregatedAPIErrors` → `KubeAggregatedAPIErrors`

### Logging chart updated and Loki improved

> We updated the logging component to the latest available FluentBit version, 1.8.13.

Reviewer 4: ```suggestion
We updated the Logging component to the latest available FluentBit version, 1.8.13.
```

## Serverless

### Node.js 12 deprecated

The base image for [Node.js 12 will be maintained only until April 2022](https://nodejs.org/en/about/releases/). Thus, following the recommendations from Node.js, we are planning to remove Node.js 12 from the list of supported runtimes soon. This basically means that Node.js 12-based Functions will continue to work, but you won’t be able to create a new Function on this runtime or edit an existing Function to run on it.

For now, we recommend that you re-configure all your Node.js 12 Functions to run on the Node.js 14 runtime.

## Service Mesh

### FIPS distroless Istio images provided by Solo

With Kyma 2.1, both [Istio control plane and data plane](https://istio.io/latest/docs/ops/deployment/architecture/) now use distroless images compliant with Federal Information Processing Standards (FIPS). [Solo](https://www.solo.io/) provides the FIPS-certified images. To learn more, read about [Distroless FIPS-compliant Istio](https://www.solo.io/blog/distroless-fips-compliant-istio/).

### Istio upgraded to 1.12.3

> With this release, we upgraded Istio from 1.11.4 to 1.12.3. For more details on the introduced changes, read the official [Istio 1.12.3 release notes](https://istio.io/latest/news/releases/1.12.x/announcing-1.12.2/).

Reviewer 4: ```suggestion
With this release, we upgraded Istio from 1.11.4 to 1.12.3. For more details on the introduced changes, read the official [Istio 1.12.3 release notes](https://istio.io/latest/news/releases/1.12.x/announcing-1.12.3/).
```

Author: 🙈 Thanks!

### Istio refactored

With Kyma 2.1, we improved the Istio component, splitting it into Istio and Istio Resources charts. The Istio component is now solely responsible for pure Istio installation. The Istio Resources chart includes Kyma-specific configuration.

> The Istio Configuration chart was renamed to Istio. 

Reviewer 4: So the Istio Configuration chart is now a part of Istio? The same that handles installation? 🤔

Author: I understand that it all is (and was) part of Istio (the component), it's just the chart name that's changed.
But maybe it's a good idea to ask @strekm.

Author: So basically, we have the Istio component with its "entourage" (chart, reconciler), which is a prerequisite, and the we also have the `istio-resources` chart, which is just that, it's not a component, and it's not required. You just won't have the dashboards if you don't install the `istio-resources` chart, but Kyma will still work correctly.

- [Observability](#observability) - Kiali and Jaeger upgraded, monitoring upgraded, metrics restructured, Grafana dashboards improved, the `alertmanager` chart revamped, logging chart updated and Loki improved
- [Serverless](#serverless) - Node.js 12 deprecated
- [Service Mesh](#service-mesh) - FIPS distroless Istio images provided by Solo, Istio refactored and upgraded to 1.12.3

## General

With release 2.1, we now officially support and test against Kubernetes 1.21.

## CLI

### `deploy` command has the dry-run option in an alpha version

<!-- TODO: waiting for input -->

### Specify components with **downloadURL** and **version**

<!-- TODO: waiting for input -->

### Generate schema for Function manifests

We improved development experience for Functions developers. We added schema generation for the `config.yaml` manifests. This, when combined with dedicated plugins in your IDE, helps you avoid mistakes via validation and auto-completion.

- If you use the VSCode IDE to develop Function code, just  use the `--vscode` option when generating the Function project:
  ```bash
  kyma init function --vscode
  ```

>   This command  creates the `.vscode` folder, which contains the schema for `config.yaml`. Install a [YAML plugin](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) in your IDE and the VSCode will automatically pick it up for validation and autocompletion.

Reviewer 6: Try to change the link here for a link directly to extension repo https://github.com/redhat-developer/vscode-yaml then we will see what job do.
